howto/finetune_lora.md

```diff
@@ -0,0 +1,88 @@
+# Finetuning with LoRA
+
+[Low-rank adaption (LoRA)](https://arxiv.org/abs/2106.09685) is a technique to approximate the update to the linear layers in a LLM with a low-rank matrix factorization. This significantly reduces the number of trainable parameters and speeds up training with little impact on the final performance of the model.
+We demonstrate this method by instruction-finetuning Lit-Parrot StableLM 3B on the [Alpaca](https://github.com/tatsu-lab/stanford_alpaca) dataset on a **single RTX 3090 (24GB) GPU**.
+
+## Preparation
+
+The steps here only need to be done once:
+
+1. Follow the instructions in the [README](../README.md) to install the dependencies.
+2. Download and convert the weights and save them in the `./checkpoints` folder as described [here](download_weights.md).
+3. Download the data and generate the instruction tuning dataset:
+
+   ```bash
+   python scripts/prepare_alpaca.py
+   ```
+
+See also: [Finetuning on an unstructured dataset](unstructured_dataset.md)
+
+## Running the finetuning
+
+```bash
+python finetune/lora.py
+```
+
+The finetuning requires at least one GPU with ~24 GB memory (RTX 3090).
+
+This script will save checkpoints periodically to the folder `out/`.
+
+
+
+## Test the model
+
+You can test the finetuned model with your own instructions by running:
+
+```bash
+python generate/lora.py --prompt "Recommend a movie to watch on the weekend."
+```
+Output:
+```
+I would recommend the movie The Martian (2015). It is a sci-fi movie starring Matt Damon that follows the story of...
+```
+
+If your GPU supports `bfloat16`, you can additionally pass `--dtype bfloat16` to bring the memory consumption down to ~14 GB.
```

Comment:
`dtype` doesn't exist, it's `precision` here (passed to fabric)

Same question about the memory requirements!

Reply:
Just checked and it's actually 11 GB only for StableLM 3B :)